src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1605,8 +1605,10 @@ static ucs_status_t uct_ib_mlx5dv_check_dc(uct_ib_device_t *dev)
         goto out;
     }
 
-    status = uct_ib_mlx5dv_qp_tmp_objs_create(dev, pd, &qp_tmp_objs);
+    status = uct_ib_mlx5dv_qp_tmp_objs_create(dev, pd, &qp_tmp_objs, 1);
     if (status != UCS_OK) {
+        /* Change status to allow other md_open() versions*/
```

Comment:
See other places in the function - we need to set status==UCS_OK.
just by jumping to out_dealloc_pd, we skip `dev->flags |= UCT_IB_DEVICE_FLAG_DC`

Reply:
It's not clear for me, why do we want to set the `UCT_IB_DEVICE_FLAG_DC` flag when we have too low memlock limit.

Reply:
we could set status==UCS_OK (instead of UNSUPPORTED) and jump to out_dealloc_pd
so UCT_IB_DEVICE_FLAG_DC  will **not** be set

Reply:
So the only place where we return error from this function is that:
```c
    pd = ibv_alloc_pd(ctx);
    if (pd == NULL) {
        ucs_error("%s: ibv_alloc_pd() failed: %m", uct_ib_device_name(dev));
        status = UCS_ERR_IO_ERROR;
        goto out;
    }
```
Can we just skip DC support in that case too and print the message on debug level? If so, then we can make `check_dc` function returning void.

Reply:
> Can we just skip DC support in that case too and print the message on debug level? If so, then we can make `check_dc` function returning void.

yes